web/src/main/java/org/springframework/security/web/header/writers/ClearSiteDataHeaderWriter.java

```diff
@@ -74,11 +74,12 @@ public void writeHeaders(HttpServletRequest request, HttpServletResponse respons
 		if (this.requestMatcher.matches(request)) {
 			if (!response.containsHeader(CLEAR_SITE_DATA_HEADER)) {
 				response.setHeader(CLEAR_SITE_DATA_HEADER, this.headerValue);
-			}
+			}			
+		} else {
```

Comment:
Instead of adding an `else` clause, will you please add a `return;` inside of the `if`? This will leave the nesting as-is.